Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
> for autodetecting the shell, do we still want to check the argv[0] of the parent process and then fall back to the value of `SHELL`? That seems to be what folks have settled on but I want to make sure.

Yes, that was also my understanding and I'm in support of this approach.

> now that `--env` is a bona fide subcommand does it make sense to make it `brew env` (and keep the `--env` for backward compatibility)?

I agree that passing additional options to that feels a bit weird, but let's stick with this command name for now. We've already renamed it from `environment` to `--env` in the past and I'm pretty hesitant to add yet another alias. (In any case, and if other maintainers feel like you, that's something that should go into a follow-up PR. Let's keep this one more focused, so that it can be merged sooner than later.)